tooling/karaf-maven-plugin/src/main/java/org/apache/karaf/tooling/RunMojo.java

```diff
@@ -153,6 +158,14 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         }
     }
 
+    void fixProjectArtifact() {
+        Artifact artifact = project.getArtifact();
+
+        if(artifact.getFile() == null || !artifact.getFile().exists()) {
+            artifact.setFile(new File(bundleArchivePath));
```

Comment:
Sorry, but I didn't understand the purpose here. `karaf:run` is supposed to be in an execution cycle, so it should not attach any artifact.